### ISSUE TITLE
Interested in making package available from Debian?

### ISSUE THREAD

**Issue by petterreinholdtsen** [piql/insight#9]
Hi

Are you interested in doing the work required to get this program into Debian?  I am a Debian developer and can help make it happen.  It is a lot easier if the upstream developer team is interested making it happen and willing to modify the program to integrate with Debian.

I have noticed a few things that will need to change for this to happen:

- configuration file handling should look somewhere relative to ~/, not in cwd, for the settings
- the reports/ direcotry should probably be stored in a fixed location relative to ~/.cache/, not in csd.
- indexing of document content should work.
- must include .desktop file for menu integration
- should include manual page

These are probably relative small changes.

I already have draft Debian build rules, which will need some cleanup before it is ready for upload.

**Comment by oleliabo** [comment]
Yes this is interesting! We are currently preparing v2 of the SW and part of that work is to package it for RPMs / RedHat Linux. I guess packaging for Debian has some similarities. So far the application has been tested on Win/Centos/OS-X, but only released for Win.

>the reports/ directory should probably be stored in a fixed location relative to ~/.cache/, not in csd.
The reports dir is currently controlled by the config file, maybe creating a default Debian config file would solve this?

>indexing of document content should work.
Yes, this is probably most complicated. We are currently using sphinx indexer. I assume this is already a Debian package. The indexer "server" is launched by the application and it feeds it with data using a sphinx-config file created by the application.  

>should include manual page
The .org file is maybe a good starting point?

**Comment by petterreinholdtsen** [comment]
I recommend rewriting the configuration system to be multilevel, allowing configuration to be read from several files, one included in the package, perhaps one site spesific, one host spesific and one user spesific.  This will allow maximum flexibility and control.  I've looked at the code but am not done with such change.   The idea is to look for settings in /usr/share/piql-insight/insight.conf, /etc/piql-insight/insight.conf and ~/.config/piql-insight/insight.conf, and use the last found setting.

As for writing a manual page, it is not too hard using for example txt2man.

I suggest we handle the missing indexing/search feature in #8, and have followed up there.

**Comment by oleliabo** [comment]
>I recommend rewriting the configuration system to be multilevel, allowing configuration to be read from several files, one included in the package, perhaps one site spesific, one host spesific and one user spesific. 

OK, maybe QSettings can provide this: https://doc.qt.io/qt-5/qsettings.html#platform-specific-notes

**Comment by petterreinholdtsen** [comment]
I noticed another platform specific hardcoding.  The code starting the mail sending process is looking for outlook.exe.  This should be generalized too.

**Comment by petterreinholdtsen** [comment]
Btw, when will the RedHat port be publicly available?  As I expect it to clean up most platform specific hardcoded values, I do not look into those myself to avoid duplicate work, and it would be useful to know when it is sensible to continue looking at this code.

**Comment by oleliabo** [comment]
Hello, I hope to release the beta of the next version in a couple of weeks. 

> I noticed another platform specific hardcoding. The code starting the mail sending process is looking for outlook.exe. This should be generalized too.

The mail app can be configured through the config file, default is perhaps outlook.

**Comment by petterreinholdtsen** [comment]
[Ole Liabø]
> The mail app can be configured through the config file, default is
> perhaps outlook.

I noticed and tried to adjust it to work with Mutt, but it do not work
if the program is to be found in PATH and need some arguments to run
(like 'xterm -e mutt' to run text mail client in a terminal).  So it
need some adjustments to work well.

-- 
Happy hacking
Petter Reinholdtsen

**Comment by petterreinholdtsen** [comment]
Hi.  Do you have a timeline for when the new version will show up?

**Comment by oleliabo** [comment]
We have picked it up again now, I hope within a week or two..

**Comment by petterreinholdtsen** [comment]
I hope so too.  In the mean time, I tried to build the wip/v1.1 branch on Debian Buster, and had to modify the qmake build rules a bit to get it working.  The attached patch need more work, but gave me a insight binary.  Unfortunately the binary do not work, but only pop up with a dialog box stating that it "Failed to load translation file: insight_nb.qm".

[qmake-debian.diff.txt](https://github.com/piql/insight/files/4768797/qmake-debian.diff.txt)

**Comment by oleliabo** [comment]
I have now merged wip/v1.1 into master and will release a new version for Windows 64bit tomorrow, since this is the current target platform for some customers. Ill try to incorperate Your patch as well then...

**Comment by oleliabo** [comment]
Latest version is a step forward at least. It is compiled and tested on Ubuntu 18.04. The `index.sh` starts the indexer, change the `insight.conf` to point to it on Linux. (some examples are provided in the config file). The missing translation file can be created using `update-translations.sh`, starting the `lupdate` Qt tool that deals with localizations.
